src/dataframerow/dataframerow.jl

```diff
@@ -270,7 +270,14 @@ Base.get(f::Base.Callable, dfr::DataFrameRow, key::ColumnIndex) =
 Base.broadcastable(::DataFrameRow) =
     throw(ArgumentError("broadcasting over `DataFrameRow`s is reserved"))
 
-Base.NamedTuple(dfr::DataFrameRow) = NamedTuple{Tuple(keys(dfr))}(values(dfr))
+function Base.NamedTuple(dfr::DataFrameRow)
+    k = Tuple(_names(dfr))
+    v = ntuple(i -> dfr[i], length(dfr))
+    pc = parentcols(index(dfr))
+    cols = _columns(parent(dfr))
+    s = ntuple(i -> eltype(cols[pc[i]]), length(dfr))
+    NamedTuple{k, s...}(v)
```

Comment:
```suggestion
    NamedTuple{k, Tuple{s...}}(v)
```
It was my mistake. Sorry for this. I have benchmarked both and it is only minimally slower than the previous version, so this seems OK:
```
julia> dfr
DataFrameRow
│ Row │ x      │ y      │
│     │ Int64? │ Int64? │
├─────┼────────┼────────┤
│ 1   │ 1      │ 2      │

julia> @benchmark NamedTuple($dfr)
BenchmarkTools.Trial: 
  memory estimate:  480 bytes
  allocs estimate:  9
  --------------
  minimum time:     814.301 ns (0.00% GC)
  median time:      824.548 ns (0.00% GC)
  mean time:        884.224 ns (3.93% GC)
  maximum time:     54.187 μs (98.25% GC)
  --------------
  samples:          10000
  evals/sample:     83

julia> @benchmark fun($dfr)
BenchmarkTools.Trial: 
  memory estimate:  496 bytes
  allocs estimate:  11
  --------------
  minimum time:     904.462 ns (0.00% GC)
  median time:      924.692 ns (0.00% GC)
  mean time:        1.032 μs (4.61% GC)
  maximum time:     126.462 μs (99.00% GC)
  --------------
  samples:          10000
  evals/sample:     39
```
(`fun` is the proposed implementation, `NamedTuple` is the old implementation)